frontend/src/scenes/insights/Insights.js

```diff
@@ -95,22 +91,6 @@ const showComparePrevious = {
     [`${ViewType.PATHS}`]: false,
 }
 
-function determineInsightType(activeView, display) {
-    if (activeView === ViewType.TRENDS || activeView === ViewType.SESSIONS) {
-        return display || ACTIONS_LINE_GRAPH_LINEAR
-    } else if (activeView === ViewType.FUNNELS) {
-        return FUNNEL_VIZ
-    } else if (activeView === ViewType.RETENTION && display !== ACTIONS_LINE_GRAPH_LINEAR) {
-        return RETENTION_TABLE
```

Comment:
Problem here is that if you're creating a dashboard item like retention, it uses `ActionsTable` to determine what chart to show, but when stored in the filter it needs to be `RETENTION_TABLE` because the dashboard item is different. That's why the special condition above exists and below for `RETENTION_GRAPH`

Reply:
Fixed. I've completely got rid of `RetentionTable` and `RetentionGraph` as options in `display`, instead just choosing them on the fly. Also fixed a few other bugs for retentiontable, like dates not working on dashboards.